frontstage/views/sign_in.py

```diff
@@ -29,27 +29,49 @@ def login():
         username = request.form.get('username')
         password = request.form.get('password')
 
-        # Creates a 'session client' to interact with OAuth2. This provides a client ID to our client that is used to
-        # interact with the server.
-        client = LegacyApplicationClient(client_id=app.config['RAS_FRONTSTAGE_CLIENT_ID'])
-
-        # Populates the request body with username and password from the user
-        client.prepare_request_body(username=username, password=password, scope=['ci.write', 'ci.read'])
-
-        # passes our 'client' to the session management object. this deals with
-        # the transactions between the OAuth2 server
-        oauth = OAuth2Session(client=client)
-        token_url = app.config['ONS_OAUTH_PROTOCOL'] + app.config['ONS_OAUTH_SERVER'] + app.config['ONS_TOKEN_ENDPOINT']
-
+        # TODO Consider moving this to a helper function.
+        # Lets get a token from the OAuth2 server
         try:
-            token = oauth.fetch_token(token_url=token_url, username=username, password=password, client_id=app.config['RAS_FRONTSTAGE_CLIENT_ID'],
-                                      client_secret=app.config['RAS_FRONTSTAGE_CLIENT_SECRET'])
-
+            token_url = app.config['ONS_OAUTH_PROTOCOL'] + app.config['ONS_OAUTH_SERVER'] + app.config['ONS_TOKEN_ENDPOINT']
+            data = {
+                'grant_type': 'password',
+                'client_id': app.config['RAS_FRONTSTAGE_CLIENT_ID'],
```

Comment:
No they are not. However I'm not sure how other implementations of an OAuth2 server will interpret the spec's. It is also being explicit on the grant_type flow by showing the 4 parameters. Any objections in keeping them in until we test against UAA?